libglusterfs/src/dict.c

```diff
@@ -693,26 +714,12 @@ dict_destroy(dict_t *this)
         return;
     }
 
-    data_pair_t *pair = this->members_list;
-    data_pair_t *prev = this->members_list;
     glusterfs_ctx_t *ctx = NULL;
     uint64_t current_max = 0;
-    uint32_t total_pairs = 0;
 
     LOCK_DESTROY(&this->lock);
 
-    while (prev) {
-        pair = pair->next;
-        data_unref(prev->value);
-        GF_FREE(prev->key);
-        if (prev != &this->free_pair) {
-            mem_put(prev);
-        } else {
-            this->free_pair.key = NULL;
-        }
-        total_pairs++;
```

Comment:
We should get this number from this->count before calling dict_clear_data and then use it. Specially if you are going to implement my previous comment.

Reply:
- agree about `this->count`
- the usage of `total_pairs` (and entire block) comes from the original imp of `dict_destroy`